onnxruntime/core/platform/posix/env.cc

```diff
@@ -143,47 +143,62 @@ class PosixThread : public EnvThread {
   PosixThread(const ORTCHAR_T* name_prefix, int index,
               unsigned (*start_address)(int id, Eigen::ThreadPoolInterface* param), Eigen::ThreadPoolInterface* param,
               const ThreadOptions& thread_options) {
-    pthread_attr_t attr;
-    int s = pthread_attr_init(&attr);
-    if (s != 0) {
-      auto[err_no, err_msg] = GetSystemError();
-      ORT_THROW("pthread_attr_init failed, error code: ", err_no, " error msg: ", err_msg);
-    }
-    if (thread_options.stack_size > 0) {
-      s = pthread_attr_setstacksize(&attr, thread_options.stack_size);
+    custom_create_thread_fn = thread_options.custom_create_thread_fn;
+    custom_thread_creation_options = thread_options.custom_thread_creation_options;
+    custom_join_thread_fn = thread_options.custom_join_thread_fn;
+
+    if (custom_create_thread_fn) {
+      hThread = (pthread_t)custom_create_thread_fn(custom_thread_creation_options, CustomThreadMain, new Param{name_prefix, index, start_address, param, thread_options});
+      if (!hThread) {
+        ORT_THROW("custom_create_thread_fn returned invalid handle."); 
```

Comment:
nit: How do we know the handle is invalid? It's probably better to say that the custom thread creation fn returned nullptr.

Reply:
We have comment to indicate the invalid case:
`* Onnxruntime will throw exception on return value of nullptr or 0, indicating that the function failed to create a thread
*/
typedef CUSTOM_THREAD_HANDLE (*CustomCreateThreadFn)(void* custom_thread_creation_options, OrtThreadWorkerFn ort_thread_worker_fn, void* worker_fn_param);
`